internal/block/block_test.go

```diff
@@ -50,19 +52,24 @@ func (testSuite *MemoryBlockTest) TestMemoryBlockWriteWithDataGreaterThanCapacit
 	mb, err := createBlock(1)
 	require.Nil(testSuite.T(), err)
 	content := []byte("hi")
-	err = mb.Write(content)
+	var n int
+	n, err = mb.Write(content)
 
 	assert.NotNil(testSuite.T(), err)
+	assert.Equal(testSuite.T(), 0, n)
 	assert.EqualError(testSuite.T(), err, outOfCapacityError)
 }
 
 func (testSuite *MemoryBlockTest) TestMemoryBlockWriteWithMultipleWrites() {
 	mb, err := createBlock(12)
 	require.Nil(testSuite.T(), err)
-	err = mb.Write([]byte("hi"))
+	var n int
+	n, err = mb.Write([]byte("hi"))
 	assert.Nil(testSuite.T(), err)
-	err = mb.Write([]byte("hello"))
+	assert.Equal(testSuite.T(), 2, n)
+	n, err = mb.Write([]byte("hello"))
```

Comment:
I think everything in the "Arrange" block should be `require` and not `assert`. The functionality is not really being tested in "Arrange" so any precondition error should be fatal for the test run which is achieved via `require`.